system_tests/utils.go

```diff
@@ -745,18 +745,16 @@ func publishAndConsumeMQTTMsg(hostname, nodePort, username, password string, ove
 	c := mqtt.NewClient(opts)
 
 	var token mqtt.Token
-	for retry := 0; retry < 5; retry++ {
-		fmt.Printf("Attempt #%d to connect using MQTT\n", retry)
+	Eventually(func() bool {
 		token = c.Connect()
 		// Waits for the network request to reach the destination and receive a response
-		Expect(token.WaitTimeout(3 * time.Second)).To(BeTrue())
+		token.WaitTimeout(3 * time.Second)
 
 		if err := token.Error(); err == nil {
-			break
+			return true
```

Comment:
https://github.com/eclipse/paho.mqtt.golang/blob/ba85050a1f239f4e954dc95920213db51f937df1/token.go#L38-L42 states

> In the case of a timeout the Token does not have an error set

Therefore, with the current logic, this expectation would succeed after a 3 seconds timeout.

Reply:
Good catch! I just pushed a new commit checking the return state of `WaitTimeout()`.